Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -472,6 +472,51 @@ public OrderTicket StopLimitOrder(Symbol symbol, decimal quantity, decimal stopP
             return Transactions.AddOrder(request);
         }
 
+        /// <summary>
+        /// Send a guaranteed VWAP order to the transaction handler
+        /// </summary>
+        /// <param name="symbol">String symbol for the asset</param>
+        /// <param name="quantity">Quantity of shares for VWAP order</param>
+        /// <param name="tag">String tag for the order (optional)</param>
+        /// <returns>Order id</returns>
+        public OrderTicket VwapOrder(Symbol symbol, int quantity, string tag = "")
+        {
+            return VwapOrder(symbol, (decimal)quantity, tag);
+        }
+
+        /// <summary>
+        /// Send a guaranteed VWAP order to the transaction handler
+        /// </summary>
+        /// <param name="symbol">String symbol for the asset</param>
+        /// <param name="quantity">Quantity of shares for VWAP order</param>
+        /// <param name="tag">String tag for the order (optional)</param>
+        /// <returns>Order id</returns>
+        public OrderTicket VwapOrder(Symbol symbol, double quantity, string tag = "")
+        {
+            return VwapOrder(symbol, (decimal)quantity, tag);
+        }
+
+        /// <summary>
+        /// Send a guaranteed VWAP order to the transaction handler
+        /// </summary>
+        /// <param name="symbol">String symbol for the asset</param>
+        /// <param name="quantity">Quantity of shares for VWAP order</param>
+        /// <param name="tag">String tag for the order (optional)</param>
+        /// <returns>Order id</returns>
+        public OrderTicket VwapOrder(Symbol symbol, decimal quantity, string tag = "")
+        {
+            var security = Securities[symbol];
+            var request = CreateSubmitOrderRequest(OrderType.Vwap, security, quantity, tag, properties: DefaultOrderProperties?.Clone());
+            var response = PreOrderChecks(request);
```

Comment:
Do we need to update `PreOrderChecks` to verify the submit time? I'm thinking in the same fashion as MOC orders.